### PR TITLE
Project Exorcism Episode 2 : Sneed's Snairs (Formerly Chuck's)

### PR DESCRIPTION
**Pull request by alsoandanswer**
New chair sprites. Nuff' said.

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!review

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 1]
```diff
@@ -55,6 +55,16 @@
 			return FALSE
 	return ..()
 
+/obj/structure/chair/fancy //base for any chair with armrests
```
Comment (alsoandanswer) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:58]:
```suggestion
/obj/structure/bed/stool/chair/fancy
```

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 2]
```diff
@@ -312,19 +160,33 @@
...
+	base_icon = "plastic_chair"
+	material_alteration = MATERIAL_ALTERATION_NAME || MATERIAL_ALTERATION_DESC
+	held_item = /obj/item/material/stool/chair/plastic
+	can_pad = FALSE
+	anchored = FALSE
 
 /obj/structure/bed/stool/chair/plastic/New(var/newloc)
 	..(newloc, MATERIAL_PLASTIC)
 
+/obj/structure/bed/stool/chair/plastic/post_buckle() //you do not want to see an angry spaceman speeding while holding dearly onto it
```
Comment (alsoandanswer) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:179]:
```suggestion
/obj/structure/bed/stool/chair/plastic/post_buckle(atom/movable/MA) //you do not want to see an angry spaceman speeding while holding dearly onto it
```

File: code/game/objects/structures/stool_bed_chair_nest/stools.dm [hunk 3]
```diff
@@ -153,6 +154,22 @@
...
 
+/obj/structure/bed/chair/stool/bamboo
+	name = "bamboo stool"
+	desc = "A makeshift bamboo stool with a rustic look."
+	icon_state = "bamboo_stool_item"
+	item_state = "bamboo_stool"
+	base_icon = "bamboo_stool"
+	icon_state = "bamboo_stool"
+	material_alteration = MATERIAL_ALTERATION_NAME || MATERIAL_ALTERATION_DESC
+	origin_type = /obj/item/chair/stool/bamboo
```
Comment (alsoandanswer) [inline code/game/objects/structures/stool_bed_chair_nest/stools.dm:165]:
```suggestion
	held_item = /obj/item/chair/stool/bamboo
```

File: code/game/objects/structures/stool_bed_chair_nest/stools.dm [hunk 4]
```diff
@@ -153,6 +154,22 @@
...
+	desc = "A makeshift bamboo stool with a rustic look."
+	icon_state = "bamboo_stool_item"
+	item_state = "bamboo_stool"
+	base_icon = "bamboo_stool"
+	icon_state = "bamboo_stool"
+	material_alteration = MATERIAL_ALTERATION_NAME || MATERIAL_ALTERATION_DESC
+	held_item = /obj/item/chair/stool/bamboo
+	can_pad = FALSE
+
+/obj/structure/bed/stool/hover/New(var/newloc)
```
Comment (alsoandanswer) [inline code/game/objects/structures/stool_bed_chair_nest/stools.dm:168]:
```suggestion
/obj/structure/bed/stool/bamboo/New(var/newloc)
```

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 5]
```diff
@@ -431,15 +303,28 @@
 	base_icon = "folding_chair"
 	origin_type = /obj/structure/bed/stool/chair/folding
 
+/obj/item/material/stool/chair/plastic
+	icon_state = "plastic_chair_item"
+	item_state = "plastic_chair"
+	base_icon = "plastic_chair"
+	origin_type = /obj/structure/bed/stool/chair/wood/wings
```
Comment (alsoandanswer) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:310]:
```suggestion
	origin_type = /obj/structure/bed/stool/chair/plastic
```

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 6]
```diff
@@ -251,7 +250,7 @@
 
 /obj/structure/bed/Move()
 	. = ..()
-	if(makes_rolling_sound)
+	if(makes_rolling_sound && has_gravity())
```
Comment (Generalcamo) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm:253]: Undocumented change (but it makes sense)

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 7]
```diff
@@ -411,6 +275,14 @@
 	force_divisor = 0.5
 	origin_type = /obj/structure/bed/stool/chair
 
+/obj/item/material/stool/chair/fancy
+	name = "fancy chair"
+	desc = "Meeting brawl essential. Now all that's missing is corporate ukulele."
```
Comment (Generalcamo) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:280]: Flow error. I don't know what this means.

File: code/game/objects/structures/stool_bed_chair_nest/sofa.dm [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+/obj/structure/bed/stool/chair/sofa/right/purple/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_PURPLE)
+
+/obj/structure/bed/stool/chair/sofa/right/violet/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_VIOLET)
+
+
+/obj/structure/bed/stool/chair/sofa/corner
+	name = "impossible sofa corner"
+	desc = "This kind of sofa shouldn't even exist at all. If you see this non-euclidean specimen, contact your station's Anti-Couch Surfing Department."
```
Comment (Generalcamo) [inline code/game/objects/structures/stool_bed_chair_nest/sofa.dm:127]: This should probably be a purely OOC description to report a bug.

File: code/game/objects/structures/stool_bed_chair_nest/stools.dm [hunk 9]
```diff
@@ -153,6 +154,22 @@
 	..(newloc, MATERIAL_SHUTTLE_SKRELL)
 	set_light(1,1,LIGHT_COLOR_CYAN)
 
+/obj/structure/bed/chair/stool/bamboo
+	name = "bamboo stool"
+	desc = "A makeshift bamboo stool with a rustic look."
```
Comment (Generalcamo) [inline code/game/objects/structures/stool_bed_chair_nest/stools.dm:159]:
```suggestion
	desc = "A makeshift bamboo stool, with a rustic look."
```

File: code/game/objects/structures/stool_bed_chair_nest/stools.dm [hunk 10]
```diff
@@ -153,6 +154,22 @@
 	..(newloc, MATERIAL_SHUTTLE_SKRELL)
 	set_light(1,1,LIGHT_COLOR_CYAN)
 
+/obj/structure/bed/chair/stool/bamboo
+	name = "bamboo stool"
```
Comment (Generalcamo) [inline code/game/objects/structures/stool_bed_chair_nest/stools.dm:158]:
```suggestion
	name = "\improper bamboo stool"
```

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 11]
```diff
@@ -312,19 +160,33 @@
 	set_light(1,1,LIGHT_COLOR_CYAN)
 
 /obj/structure/bed/stool/chair/office/hover/command
-	icon_state = "hover_command_preview"
+	icon_state = "hover_command"
 	base_icon = "hover_command"
 
-// Chair types
 /obj/structure/bed/stool/chair/plastic
+	name = "plastic chair"
```
Comment (Generalcamo) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:167]:
```suggestion
	name = "\improper plastic chair"
```

**Review by Generalcamo — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/stool_bed_chair_nest/stools.dm (hunk 9 above)
Comment (Llywelwyn) [reply code/game/objects/structures/stool_bed_chair_nest/stools.dm:159]: either is correct here, personally feel like no comma is better for sentence flow

File: code/game/objects/structures/stool_bed_chair_nest/sofa.dm (hunk 8 above)
Comment (alsoandanswer) [reply code/game/objects/structures/stool_bed_chair_nest/sofa.dm:127]: This thing shouldn't be obtainable in game in the first place anyway

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm (hunk 7 above)
Comment (alsoandanswer) [reply code/game/objects/structures/stool_bed_chair_nest/chairs.dm:280]: It's just the normal chair description modified for humorous purposes.

**Review by NonQueueingMatt — APPROVED** [review]

**Comment by NonQueueingMatt** [comment]
merged in to other PR

**Pull request closed without merging**